File: solutions/c/rna-transcription/1/src/rna_transcription.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "rna_transcription.h"
char *to_rna(const char *dna)
{
    char *buffer = strndup(dna, strlen(dna));

    for (char *c = buffer; *c != '\0'; c++)
    {
        switch (*c)
        {
        case 'G':
            *c = 'C';
            continue;
        case 'C':
            *c = 'G';
            continue;
        case 'T':
            *c = 'A';
            continue;
        case 'A':
            *c = 'U';
            continue;
        default:
            return NULL;
        }
    }

    return buffer;
}

File: solutions/php/rna-transcription/1/RnaTranscription.php
<?php

declare(strict_types=1);

const dnaToRna = [
    "G" => "C",
    "C" => "G",
    "T" => "A",
    "A" => "U",
];

function toRna(string $dna): string
{
    $len = strlen($dna);
    $rna = "";

    for ($i = 0; $i < $len; $i++){
        $rna .= dnaToRna[$dna[$i]];
    }

    return $rna;
}
